NitroxPatcher/Patches/Dynamic/SpawnOnKill_OnKill_Patch.cs

```diff
@@ -0,0 +1,58 @@
+using HarmonyLib;
+using NitroxClient.Communication.Abstract;
+using NitroxClient.GameLogic;
+using NitroxClient.MonoBehaviours;
+using NitroxModel.DataStructures;
+using NitroxModel.Helper;
+using NitroxModel.Packets;
+using NitroxPatcher.PatternMatching;
+using System.Collections.Generic;
+using System.Reflection;
+using UnityEngine;
+using static System.Reflection.Emit.OpCodes;
+
+namespace NitroxPatcher.Patches.Dynamic;
+
+/// <summary>
+/// Synchronizes entities that Spawn something when they are killed, e.g. Coral Disks.
+/// </summary>
+public class SpawnOnKill_OnKill_Patch : NitroxPatch, IDynamicPatch
+{
+    public static readonly MethodInfo TARGET_METHOD = Reflect.Method((SpawnOnKill t) => t.OnKill());
+
+    private static readonly InstructionsPattern spawnInstanceOnKillPattern = new()
+    {
+        Reflect.Method(() => Object.Instantiate(default(GameObject), default(Vector3), default(Quaternion))),
+        { Stloc_0, "DropOnKillInstance" },
+        Ldarg_0,
+    };
+
+    public static IEnumerable<CodeInstruction> Transpiler(MethodBase original, IEnumerable<CodeInstruction> instructions)
+    {
+        static IEnumerable<CodeInstruction> InsertCallbackCall(string label, CodeInstruction _)
+        {
+            switch (label)
+            {
+                case "DropOnKillInstance":
+                    yield return new(Ldarg_0);
+                    yield return new(Ldloc_0);
+                    yield return new(Call, Reflect.Method(() => Callback(default, default)));
+                    break;
+            }
+        }
+
+        return instructions.Transform(spawnInstanceOnKillPattern, InsertCallbackCall);
+    }
+
+    private static void Callback(SpawnOnKill spawnOnKill, GameObject spawningItem)
+    {
+        NitroxId destroyedEntityId = NitroxEntity.GetId(spawnOnKill.gameObject);
+        Resolve<IPacketSender>().Send(new EntityDestroyed(destroyedEntityId));
```

Comment:
where went `NitroxEntity.SetNewId(spawningItem, destroyedEntityId)` ???

Reply:
We shouldn't re-use the old id because it represents a completely different type of object. Instead, we should set a new id to spawningItem.  Right now this is happening implicitly inside `Dropped` when it calls `GetId`

Reply:
Ok, fine for me 👍